client/packages/coldchain/src/Equipment/ImportAsset/EquipmentImportModal.tsx

```diff
@@ -0,0 +1,307 @@
+  FileUtils,
+} from '@openmsupply-client/common';
+import { useTranslation } from '@common/intl';
+import { AssetFragment, useAssets } from '../api';
+import { importEquipmentToCsv } from '../utils';
+import {
+  AssetCatalogueItemFragment,
+  useAssetData,
+} from '@openmsupply-client/system';
+interface EquipmentImportModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+enum Tabs {
+  Upload = 'Upload',
+  Review = 'Review',
+  Import = 'Import',
+}
+
+export type ImportRow = {
+  assetNumber: string;
+  catalogueItemCode: string | null | undefined;
+  id: string;
+  notes: string;
+  errorMessage: string;
+  isUpdate: boolean;
+};
+
+export const toInsertEquipmentInput = (
+  row: ImportRow,
+  catalogueItemData: AssetCatalogueItemFragment[] | undefined
+): AssetFragment => ({
+  assetNumber: row.assetNumber,
+  catalogueItemId: catalogueItemData
+    ?.filter(
+      (item: { code: string | null | undefined }) =>
+        item.code == row.catalogueItemCode
+    )
+    ?.map((item: { id: string }) => item.id)
+    .pop(),
+  id: row.id,
+  notes: row.notes,
+  __typename: 'AssetNode',
+  createdDatetime: undefined,
+  modifiedDatetime: undefined,
+});
+
+export const toUpdateEquipmentInput = (
+  row: ImportRow,
+  catalogueItemData: AssetCatalogueItemFragment[] | undefined
+): AssetFragment => ({
+  assetNumber: row.assetNumber,
+  catalogueItemId: catalogueItemData
+    ?.filter(
+      (item: { code: string | null | undefined }) =>
+        item.code == row.catalogueItemCode
+    )
+    ?.map((item: { id: string }) => item.id)
+    .pop(),
+  id: row.id,
+  // Assigning default values here as the parser in the API will ignore.
+  // Better type management would be gelpful here
+  __typename: 'AssetNode',
+  createdDatetime: undefined,
+  modifiedDatetime: undefined,
+});
+
+export const EquipmentImportModal: FC<EquipmentImportModalProps> = ({
+  isOpen,
+  onClose,
+}) => {
+  const t = useTranslation('coldchain');
+  const { success } = useNotification();
+  const { currentTab, onChangeTab } = useTabs(Tabs.Upload);
+  const [activeStep, setActiveStep] = useState(0);
+  const { Modal } = useDialog({ isOpen, onClose });
+
+  const [errorMessage, setErrorMessage] = useState<string>(() => '');
+  const [importProgress, setImportProgress] = useState(0);
+  const [importErrorCount, setImportErrorCount] = useState(0);
+  const {
+    data: catalogueItemData,
+    fetchAsync,
+    isLoading,
+  } = useAssetData.document.listAll();
+
+  const { mutateAsync: insertAssets } = useAssets.document.insert();
+  const { mutateAsync: updateAssets } = useAssets.document.update();
+
+  const [bufferedEquipment, setBufferedEquipment] = useState<ImportRow[]>(
+    () => []
+  );
+
+  useEffect(() => {
+    fetchAsync();
+  }, [fetchAsync]);
+
+  const csvExport = async () => {
+    const csv = importEquipmentToCsv(
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      bufferedEquipment.map((row: ImportRow): any => {
```

Comment:
I've disabled linting in a few places which I wasn't sure about including this one. 
Not sure if the data parsing can manage better typing?